packages/query/src/utils.ts

```diff
@@ -141,3 +141,22 @@ export const getHasSignal = ({
   verb: Verbs;
   overrideQuerySignal?: boolean;
 }) => overrideQuerySignal && (!getIsBodyVerb(verb) || verb === Verbs.POST);
+
+// Convert "param: Type" to "param?: Type" for queryKey functions
+// to enable cache invalidation without type assertion
+export const makeParamsOptional = (params: string) => {
```

Comment:
Currently, we are not strictly extracting functions into utils.
There are also benefits to having functions exposed in the files you use, closer to the processing they are used in.
What was the reason for moving them here now?